backend/core/utils.py

```
@@ -277,3 +281,429 @@
+        weeks_of_month = schedule.get("weeks_of_month", [])
+
+        if not months_of_year and not days_of_week and not weeks_of_month:
+            return base_date + rd.relativedelta(years=interval)
+
+        target_year = base_date.year
+
+        # If we're past all months in current year, move to next year
+        if months_of_year and base_date.month > max(months_of_year):
+            target_year += interval
+        elif base_date.month == 12:  # End of year case
+            target_year += interval
```

Ruff (0.8.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)